src/utils/helpers.test.ts

```
@@ -0,0 +1,1627 @@
+
+    it('should handle deeply nested arrays', () => {
+      expect(flattenArray([1, [2, [3, [4, [5]]]]])).toEqual([1, 2, 3, 4, 5]);
+    });
+
+    it('should handle mixed types', () => {
+      expect(flattenArray([1, ['a', [true, [null]]]])).toEqual([1, 'a', true, null]);
+    });
+
+    it('should handle sparse arrays', () => {
+      const sparse = [1, , 3, [4, , 6]];
+      const flattened = flattenArray(sparse);
```

Biome (1.9.4)

[error] This array contains an empty slot.

Unsafe fix: Replace hole with undefined


(lint/suspicious/noSparseArray)